pkg/logs/decoder/line_handler.go

```diff
@@ -246,3 +253,210 @@ func (h *MultiLineHandler) sendBuffer() {
 		h.outputChan <- NewMessage(content, h.status, h.linesLen, h.timestamp)
 	}
 }
+
+type scoredPattern struct {
+	score  int
+	regexp *regexp.Regexp
+}
+
+// DetectedPattern is a container to safely access a detected multi line pattern
+type DetectedPattern struct {
+	sync.Mutex
+	pattern *regexp.Regexp
+}
+
+// Set sets the pattern
+func (d *DetectedPattern) Set(pattern *regexp.Regexp) {
+	d.Lock()
+	defer d.Unlock()
+	d.pattern = pattern
+}
+
+// Get gets the pattern
+func (d *DetectedPattern) Get() *regexp.Regexp {
+	d.Lock()
+	defer d.Unlock()
+	return d.pattern
+}
+
+// AutoMultilineHandler can switch from single to multiline handler if upon the occurrence
+// of a stable pattern at the begginning of the process
+type AutoMultilineHandler struct {
+	multiLineHandler  *MultiLineHandler
+	singleLineHandler *SingleLineHandler
+	inputChan         chan *Message
+	outputChan        chan *Message
+	flipChan          chan struct{}
```

Comment:
what does this property do?

Reply:
It's used to stop the auto multiline handlers main loop and halt it's parsing when it switches over to a mutli line handler. I can add a comment or rename it if it's unclear.

Reply:
Got it.  A comment would be useful.  It looks like this is only used within the `run` goroutine, where `run` -> `processAndTry` -> `switchToMultilineHandler`.  Could this be replaced with a boolean that's consulted in the loop in `run`?

Reply:
Yep good point. I changed it to `isRunning` which is easier to understand.